### ISSUE TITLE
aws_db_parameter_group parameter updates not working

### ISSUE THREAD

**Issue by willthames** [terraform-providers/terraform-provider-aws#13551]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.12.26
+ provider.aws v2.64.0
+ provider.dns v2.2.0
+ provider.template v2.1.2

### Affected Resource(s)

* aws_db_parameter_group

### Terraform Configuration Files

```hcl
resource "aws_db_parameter_group" "postgres_10" {
  name        = "psql-pg-test-10"
  family      = "postgres10"
  description = "Postgres parameter group test"

  lifecycle {
    create_before_destroy = true
  }

  parameter {
    name         = "rds.pg_stat_ramdisk_size"
    value        = 256
    apply_method = "pending-reboot"
  }
}
```

### Panic Output

N/A

### Expected Behavior

Parameter group value is updated 

### Actual Behavior

Parameter group remains the same

The plan output looks like this:
```
      - parameter {
          - apply_method = "pending-reboot" -> null
          - name         = "rds.pg_stat_ramdisk_size" -> null
          - value        = "0" -> null
        }
      + parameter {
          + apply_method = "pending-reboot"
          + name         = "rds.pg_stat_ramdisk_size"
          + value        = "256"
        }
```

Reading the code, it sees this a removal of an old parameter and the addition of a new parameter, and then modifies the parameter group with the new value, and then resets the old parameter back to the existing value. Even swapping the order of those two operations would likely fix this issue

### Steps to Reproduce

1. Create an empty parameter group using terraform
2. Update the parameter group to use `rds.pg_stat_ramdisk_size = 256`

### Important Factoids

N/A

### References

* #12742 looks similar

**Comment by kevincormier-toast** [comment]
I've been running into the same issue and have been digging into the code.  It looks like the problem is here:
(https://github.com/terraform-providers/terraform-provider-aws/blob/eb6594f20ba09ab8a831869dda32c2d0b63b4d84/aws/resource_aws_rds_cluster_parameter_group.go#L225)

`Differences` compares map keys without checking values and returns the set of values in the original state that aren't in the new state.  Unfortunately, for parameter groups the keys for the map are hashes of the entire parameter (name, value, etc).  Because of this, it never shows up as an update.  It's always a removal and a re-add.  Given the order of the operations (update, then reset), this doesn't ever work!

Changing the order would probably work, but it would still be a bit ugly.  The best solution would be to replace the Differences function with something that was smart enough to only reset the values that have to be reset!

I'll try to put together a PR for this at some point, but wanted to leave this info just in case I don't get the chance or someone is able to beat me to it!

**Comment by gkop** [comment]
If you're blocked by this and are OK giving up ability to reset params, you might try `2.44`, which worked for me. I found it by looking for the latest version that didn't include https://github.com/terraform-providers/terraform-provider-aws/pull/11540 .

**Comment by forkinme** [comment]
This appears to be a duplicate of #11846 which has an associated PR #12112 under review

**Comment by anGie44** [comment]
This fix for this bug has been merged and release with `v3.2.0` of the Terraform AWS Provider, likely out this Thursday.
